pywbemtools/pywbemcli/_cmd_statistics.py

```diff
@@ -0,0 +1,347 @@
+    try:
+        conn.ModifyInstance(objmgr, PropertyList=[OBJMGR_STAT_PROPERTY_NAME],
+                            IncludeQualifiers=False)
+    except Error as ce:
+        raise click.ClickException('Modification of GatherStasticalData '
+                                   'property on server failed {}'.format(ce))
+
+    modified_objmgr = conn.GetInstance(objmgr.path)
+
+    if modified_objmgr[OBJMGR_STAT_PROPERTY_NAME] != desired_state:
+        raise click.ClickException(
+            'Modification failed classname {} on server namespace {}'.
+            format(OBJMGR_CLN, interop_ns))
+
+    click.echo("Server GatherStatisticalData set {}". format(desired_state))
+
+
+def test_stats_enabled(context):
+    """
+    Execute ClickException of timestats not enabled
+    """
+    if not context.timestats:
+        raise click.ClickException(
+            'Pywbemcli time statistics not enabled. Use general options '
+            '-T/--timestats to enable statistics.')
+
+###############################################################
+#         Statistics cmds action functions
+###############################################################
+
+
+def cmd_statistics_server_on(context):
+    """
+    Attempt to enable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, True)
+
+
+def cmd_statistics_server_off(context):
+    """
+    Attempt to disable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, False)
+
+
+def cmd_statistics_status(context):
+    """
+    Show current status of both local and server statistics
+    """
+    test_conn_exists(context)
+
+    objmgr = get_objmgr_inst(context)
+    status = objmgr[OBJMGR_STAT_PROPERTY_NAME]
+
+    local_stats = context.timestats
+    click.echo("Statistics state client={}; Server={}".format(local_stats,
+                                                              status))
+
+
+def cmd_statistics_reset(context):
+    """
+    Resets the client statistics. This command does not impact the the
+    statistics coming from the server.
+    """
+    test_conn_exists(context)
+    test_stats_enabled(context)
+
+    context.conn.statistics.reset()
+
+
+def cmd_statistics_show(context):
+    """
+    If the statistics are enabled in the local context, display them
+    """
+    test_conn_exists(context)
+    test_stats_enabled(context)
+
+    click.echo(context.format_statistics(context.conn.statistics, context))
+
+
+# pylint: disable=unused-argument
+def cmd_statistics_server_show(context):
+    """
+    If server statistics are enabled gather them from the server and display.
+
+    This command may fail if the WBEM server has not implemented the
+    gathering of detailed statistics.
+    """
+    click.ClickException("This command incomplete. See issue # 895.")
```

Comment:
Suggest to change to "This command is not implemented. See pywbemtools issue #895.".

Reply:
Done